app/models/category.rb

```diff
@@ -0,0 +1,7 @@
+class Category < ApplicationRecord
+  belongs_to :user, foreign_key: 'user_id'
+  has_many :category_transactions, dependent: :destroy
```

Comment:
Required

- As you used the many through association, it's required to have two association types, one for the main model which is the transaction, and one for the third model that connects the category and transaction through it. Currently, you're missing the has many transactions through category transaction association

Reply:
Added the association type